Update I2S sample files:

waveletboy opened the pull request:
1. Correct original BBC files to 2ch, 32bit, 16khz
2. Add sample files for 2ch, 16bit, 48khz
3. Fix the original issue on right channel for BBC files. Generated wav files can be played on both channels.
4. Generate wav files for 2ch 16bit 48khz samples

Note: to generate correct wav file need to adjust functions wav_header() and wav_sample in pd.py in i2s decoder.

waveletboy committed:
Update I2S sample files:

1. Correct original BBC files to 2ch, 32bit, 16khz
2. Add sample files for 2ch, 16bit, 48khz
3. Fix the original issue on right channel for BBC files. Generated wav files can be played on both channels.
4. Generate wav files for 2ch 16bit 48khz samples

Note: to generate correct wav file need to adjust functions wav_header() and wav_sample in pd.py in i2s decoder.

uwehermann commented:
Hi, thanks for the PR! A few suggestions for changes:

* Please drop the .wav files, only add .sr files to sigrok-dumps.

* Please only use public domain audio files or files with explicitly mentioned free-ish license (PD, CC-BY or such). Also mention the source URL for the files as well as the license in the README.

A few random suggestions for public domain audio:
https://freesound.org/people/theojt/sounds/511195/
https://freesound.org/people/tnorr6969/sounds/511174/
https://freesound.org/people/theojt/sounds/510949/

(There are many other sources, of course)

The existing BBC file is very short, but we also don't have a clear license there AFAIK. Feel free to drop that one and replace it with only new files created by you, using free source material.

Thanks!

waveletboy committed:
Update I2S sample files:

1. Remove original BBC files as unclear license.
2. Add sample files for 2ch, 16bit, 48khz from freesound
3. Generate wav files for 2ch 16bit 48khz samples

Note: to generate correct wav file need to adjust functions wav_header() and wav_sample in pd.py in i2s decoder.

waveletboy commented:
Hi Uwe,

I removed original BBC file and regenerate 2 sample files based on freesound. 
However, I want to keep decoded wave files because they can help to understand the I2S->wav decoding. User can compare their decoding output with sample wav files. Please consider. 

Thanks!

waveletboy commented:
Sorry for spamming here. Is second commit ok?

uwehermann commented:
Looks OK (will do some minor README cosmetics, but that's not a big deal), but please make the file snippets a lot smaller. Ideally < 1MB or 500kB or such per file. This makes sigrok-dumps and also sigrok-test (which will need updates as well) unnecessarily huge otherwise.

r4d10n commented:
Very useful example dumps to test decoding of I2S audio... 

Using sigrok-cli and aplay, one can test live continuous decodes. Aplay can be used in RAW mode to ignore WAVE header inaccuracies.

`sigrok-cli -i 2ch-16bit-48khz_01.sr -P i2s:sck=SCK:ws=WS:sd=SD -B i2s | aplay -t raw -f S16_LE -r 48000 -c 2`

endolith commented:
I have a lot of traces of I2S if you need more examples of streams starting and stopping, etc.
